### PR TITLE
Fix removal of translations from transifex

### PR DESCRIPTION
**Pull request by emord**
https://app.asana.com/0/671300723874076/946351016425500/f

Fixes a few bugs and adds tests:

- Removes validation, help and hint messages that should have been skipped
- Removes choices from select questions that should have been skipped
- If two questions share the same label, then the label is sent to transifex if any of the questions should be

### CODE REVIEW

**Comment by emord** [comment]
> > If two questions share the same label
> How is that even possible?

Its used to be very common, and I found it happening several times in ICDS forms last night. Its often used for repeated yes/no questions so that the resulting XML is smaller. In form builder under the "advanced" tab, there is an option to uncheck "auto" which means that you can specify the itext's id manually. With that you can set two questions to have the same itext id, which means that it would share the same translations.

This can also be done automatically by other tools that users use to import XML to commcare.

> Taking a step back here. Can we instead handle this here? I believe it would be better to not complicate this by adding references for each time we need to exclude something instead just exclude everything that comes under a question that has been marked to skip transifex. And when we don't need this in future, it would be easier and less code to remove.

I agree that would be good if these sort of instances were under the same umbrella of a question, but that's not how the XML or that code is structured. That code is iterating through the [translation nodes](https://github.com/dimagi/commcare-hq/blob/210b0dec9088ef7d970022fc900e5fe41b9e3ab1/corehq/apps/translations/app_translations.py#L437-L456), but in the form builder, you must mark these comments in relation to the question. And as said above that's not a 1-to-1 mapping.

You can check out the test, but these are stored in separate parts of the XML. [Validation message reference is in the bind node](https://github.com/dimagi/commcare-hq/blob/0243a58b456f5335c8ce9750f04082105ffec637/corehq/apps/translations/tests/data/transifex_blacklist.xml#L18). [Other references are in the body](https://github.com/dimagi/commcare-hq/blob/0243a58b456f5335c8ce9750f04082105ffec637/corehq/apps/translations/tests/data/transifex_blacklist.xml#L88-L93) and this is where that [code is stepping through](https://github.com/dimagi/commcare-hq/blob/0243a58b456f5335c8ce9750f04082105ffec637/corehq/apps/translations/tests/data/transifex_blacklist.xml#L23)

It would be great to step back and think of translations from the ground up, instead of it being a patchwork that's had bugs fixed over the years, but that's out of scope of this PR

**Comment by mkangia** [comment]
> In form builder under the "advanced" tab, there is an option to uncheck "auto" which means that you can specify the itext's id manually

Ya thats one hidden thing. Priyanka could not understand why translating one label modified all others and then found out about this and i think its enabled by default but then she disabled it. If you seeing it enabled then it can cause the same confusion at other places, can you flag that to ICDS app building team so that they can disable it or at least know that something like this is happening.

> I agree that would be good if these sort of instances were under the same umbrella of a question

Eventually Vellum connects everything related to a question under the question itself. Isn't there a way we can do that here as well?

> It would be great to step back and think of translations from the ground up, instead of it being a patchwork that's had bugs fixed over the years, but that's out of scope of this PR

happy to improve the translations code base. Do you need things off hand that can be improved?

> As a future refactor, maybe the code you linked could be generalized into "translations by question" instead of "all translations" and we can build the XLS file from that, instead of building the XLS file from all translations. Thoughts?

You mean modifying the form XML to not store all translations under one tag for a language but store if per question? That sounds like a big change that would also need migrating all form xmls into this new format. But if you have an idea of how that can be done, i can try to make it happen.

**Comment by mkangia** [comment]
@emord i see that this PR is time sensitive. If you are confident about the way things are handled in PR feel free to go ahead with it, though i believe we should reconsider it for better handling, covering all cases.

File: corehq/apps/reports/tests/readable_forms/mismatched_group_hierarchy.result.yaml [hunk 1]
```diff
@@ -7,7 +7,8 @@
   label: You first need to log a facility before you can record information.
   label_ref: need_facility_trigger-label
   options:
-  - {label: Go back and exit form, value: exit_form}
+  - {label: Go back and exit form, label_ref: need_facility_trigger-exit_form-label,
```
Comment (mkangia) [inline corehq/apps/reports/tests/readable_forms/mismatched_group_hierarchy.result.yaml]: You had to manually add all these?? :/

**Review by mkangia — APPROVED** [review]

File: corehq/apps/reports/tests/readable_forms/mismatched_group_hierarchy.result.yaml (hunk 1 above)
Comment (emord) [reply corehq/apps/reports/tests/readable_forms/mismatched_group_hierarchy.result.yaml]: It's not totally manual. when the test fails it gives you some message that shows you how to print out hte correct yaml file https://github.com/dimagi/commcare-hq/blob/3a75291943361657e39af72270b3068743f79928/corehq/apps/reports/tests/test_readable_formdata.py#L273-L281

**Pull request merged**